tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -230,30 +230,53 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext {
     }
   }
 
-  private def mkSelect(setId: Int, elemId: Int): ExprSort = {
-    val set = cellCache(setId).head._1
+  private def mkSelect(arrayId: Int, elemId: Int): ExprSort = {
+    val array = cellCache(arrayId).head._1
     val elem = cellCache(elemId).head._1
 
+    z3context.mkSelect(array.asInstanceOf[ArrayExpr[Sort, Sort]], elem.asInstanceOf[Expr[Sort]]).asInstanceOf[ExprSort]
+  }
+
+  private def mkNestedSelect(outerArrayId: Int, innerArrayId: Int, elemId: Int): ExprSort = {
+    val outerArray = cellCache(outerArrayId).head._1
+    val innerArray = cellCache(innerArrayId).head._1
+    val elem = cellCache(elemId).head._1
+
+    val innerSelect = z3context
+      .mkSelect(innerArray.asInstanceOf[ArrayExpr[Sort, Sort]], elem.asInstanceOf[Expr[Sort]])
+      .asInstanceOf[ExprSort]
     z3context
-      .mkSelect(set.asInstanceOf[ArrayExpr[Sort, BoolSort]], elem.asInstanceOf[Expr[Sort]])
+      .mkSelect(outerArray.asInstanceOf[ArrayExpr[Sort, Sort]], innerSelect.asInstanceOf[Expr[Sort]])
       .asInstanceOf[ExprSort]
   }
 
-  private def mkStore(setId: Int, elemId: Int): ExprSort = {
-    val (set, setT, _) = cellCache(setId).head
+  private def mkStore(arrayId: Int, elemId: Int, IndexId: Int = 1): ExprSort = {
+    val (array, arrayT, _) = cellCache(arrayId).head
     val (elem, elemT, _) = cellCache(elemId).head
-
-    val edgeName = s"in_${elemT.signature}${elemId}_${setT.signature}$setId"
-    log(s";; declare edge inclusion $edgeName")
-    val updatedSet = updateSetConst(setId)
-
-    val store = z3context.mkStore(
-        set.asInstanceOf[Expr[ArraySort[Sort, BoolSort]]], elem.asInstanceOf[Expr[Sort]], z3context.mkTrue())
-    val eqStore = z3context.mkEq(updatedSet, store)
+    val (index, indexT, _) = cellCache(IndexId).head // Index 1 caches the value true
+
+    val updatedArray = updateArrayConst(arrayId)
+    val store = IndexId match {
+      case 1 =>
```

Comment:
Should we use `Arena.cellTrue().id` here?

Reply:
`cellTrue()` does not seem to be globally available.